writers/batchwriter/batchwriter.go

```diff
@@ -122,8 +122,8 @@ func (w *BatchWriter) Close(context.Context) error {
 func (w *BatchWriter) worker(ctx context.Context, tableName string, ch <-chan *message.WriteInsert, flush <-chan chan bool) {
 	sizeBytes := int64(0)
 	resources := make([]*message.WriteInsert, 0, w.batchSize)
-	tick, done := writers.NewTicker(w.batchTimeout)
-	defer done()
+	ticker := writers.NewTicker(w.batchTimeout)
+	defer ticker.Stop()
```

Comment:
`ticker.Chan()` should also be closed in the defer. Could maybe add it in a `Stop()` wrapper since we only stop on defer and there would be no erroneous tick read in the `select`:

> Stop turns off a ticker. After Stop, no more ticks will be sent.
> Stop does not close the channel, to prevent a concurrent goroutine
> reading from the channel from seeing an erroneous "tick".